plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/members/ComposedMemberScope.java

```diff
@@ -219,16 +229,44 @@ private TMember getOrCreateComposedMember(String memberName) {
 	}
 
 	/**
+	 * Returns the composed member cache for the given ComposedTypeRef, creating it if it does not exist yet. Returns
+	 * <code>null</code> if a cache could not be created, because the given type reference is not contained in an
+	 * N4JSResource or this resource does not have a TModule.
+	 */
+	private ComposedMemberCache getOrCreateComposedMemberCache(ComposedTypeRef ctr) {
+		final ComposedTypeRef cacheHolder = getCacheHolder(ctr);
```

Comment:
the variable name `cacheHolder` and method name `getCacheHolder` are confusing in this context. We should rename them to some thing like `originalComposedTypeRef` or something like that.